### PR TITLE
[Parseable output] Implement SignalledMessage

### PR DESCRIPTION
**Pull request by sjavora**
Implements `SignalledMessage` as another kind of `ParsableMessage`.

### CODE REVIEW

File: Sources/SwiftDriver/Driver/ToolExecutionDelegate.swift [hunk 1]
```diff
@@ -58,18 +58,19 @@ public struct ToolExecutionDelegate: JobExecutorDelegate {
...
-        let message = ParsableMessage.finishedMessage(name: job.kind.rawValue, msg: finishedMessage)
-        emit(message)
+        let finishedMessage = FinishedMessage(exitStatus: Int(code), pid: pid, output: output)
+        message = ParsableMessage.finishedMessage(name: job.kind.rawValue, msg: finishedMessage)
 
-      case .signalled:
-        // FIXME: Implement this.
-        break
+      case .signalled(let signal):
+        let signalledMessage = SignalledMessage(pid: pid, output: output, errorMessage: output ?? "", signal: Int(signal))
```
Comment (sjavora) [inline Sources/SwiftDriver/Driver/ToolExecutionDelegate.swift]: Is this right? I tried looking through the existing driver, but got lost somewhere around [here](https://github.com/apple/swift/blob/632399e0f8345e9c28d03b26252b5a30078489fc/lib/Driver/Compilation.cpp#L1131). I couldn't find any other errors in the `result`.

Comment (DougGregor) [reply Sources/SwiftDriver/Driver/ToolExecutionDelegate.swift]: It looks like the existing driver is using [`strsignal`](http://man7.org/linux/man-pages/man3/strsignal.3.html) to get a string representation of the error message.

Comment (sjavora) [reply Sources/SwiftDriver/Driver/ToolExecutionDelegate.swift]: Ah I see it now in TaskQueue. Implemented! Hope those new imports are correct.

**Comment by sjavora** [comment]
> have you tried this out through a consumer of parsable output, like Xcode?

I've gotten this output: `error: Illegal instruction: 4 (in target 'TSCLibc' from project 'swift-driver')`

It seems consistent with the existing driver.

**Comment by DougGregor** [comment]
> > have you tried this out through a consumer of parsable output, like Xcode?
> 
> I've gotten this output: `error: Illegal instruction: 4 (in target 'TSCLibc' from project 'swift-driver')`
> 
> It seems consistent with the existing driver.

Great! Before your PR, it would silently fail in Xcode with no message.

**Comment by DougGregor** [comment]
@swift-ci please test

**Pull request merged**